### DISCUSSION TITLE
getServerSession() doesn't pass session to client component when deployed to Vercel

### DISCUSSION THREAD

**Discussion by jhughes982** [nextauthjs/next-auth#8789] [Help]
### Summary

I am currently fetching the next-auth session in a server component using `getServerSession(options)`. I then pass that session to a client component. The client component displays navlinks that shows `Sign Out` if a session is found and `Sign In` if a session is `null`.

When I first sign in, the navlink successfully changes from `Sign In` to `Sign Out`. But then if I navigate to a different page (e.g. `/Home`) the navlink then changes back to `Sign In`, even though the session is still found in the cookies. Interestingly, if I manually refresh the page the navlink goes back to `Sign Out` until I again navigate to a different page, at which point it switches back to `Sign In` even though the session cookies are still present.

So what I think is happening here (I could be wrong) is that my parent server component successfully gets the session on first page load and then passes that session to the child client component. But then when I navigate using the links on my site (using the Link component from `next/link`) the session is no longer getting passed to my child component, presumably because of some caching behaviour?

So I am looking for a solution where my parent server component gets the session every time I navigate using the links on my page. 

Weirdly, everything works fine on my dev environment and the `Sign Out` navlink persists correctly. It even works fine when I deploy to Render.com. But when I deploy to Vercel I get the behaviour as described above, so it seems to be specific to the Vercel deployment environment.

Note, i am using the new `app` directory.

Parent component:
```
import { options } from '../../app/api/auth/[...nextauth]/options';
import { getServerSession } from 'next-auth/next';

import NavLinks from './NavLinks';

import classes from './Navbar.module.css';

async function Navbar() {
  const session = await getServerSession(options);

  console.log('navbar session', session);

  return (
    <header className={classes.header}>
      <NavLinks session={session} />
    </header>
  );
}

export default Navbar;

```

Child component:
```
'use client';
import { signOut } from 'next-auth/react';
import Link from 'next/link';

import classes from './Navbar.module.css';

function NavLinks(props) {
  const { session } = props;

  console.log('navlinks session', session);

  const signOutHandler = () => {
    signOut({ callbackUrl: '/' });
  };

  return (
      <nav className={classes.nav}>
        <ul>
          <li>
            <Link href="/">Home</Link>
          </li>

          {!session && (
            <li>
              <Link href="/auth/signin">Sign In</Link>
            </li>
          )}

          {session && (
            <li>
              <div
                className={classes['signout-btn']}
                onClick={() => signOutHandler()}
              >
                Sign Out
              </div>
            </li>
          )}
        </ul>
      </nav>
  );
}

export default NavLinks;

```

**Comment by krthush** [comment]
I'm having the exact same issue as well - I created another discussion about it so I could share our code / experience as well: https://github.com/nextauthjs/next-auth/discussions/8806#discussion-5707991

@jhughes982 did you manage to find a fix for this?

**Reply by jhughes982** [reply]
I haven't found a solution to this yet. For now I am forced to use the `useSession()` hook inside the client component directly. But this has the disadvantage of having a temporary loading state while the session is being set, which means there is a flash on the screen before the content is rendered correctly.

**Reply by krthush** [reply]
Makes sense, in our app I was using the getServerSession in the layout file so not easy to switch to a client component without having to restructure a lot of code. It is doable but I'd rather avoid it :(
Even then after that like you said you'd starting running into UX issues like the flashing / loading screens
It seems like this issue only cropped up recently, since our app has been in production for a few months and was working fine before 1-2 weeks ago.
